### ISSUE TITLE
set customer language 

### ISSUE THREAD

**Issue by MarcusWolschon** [sumup/sumup-android-sdk#30]
According to support there is no way yet to have the SumUp card reader ask the customer to enter pin, wait, sign, ... in another language. 
There should be a way towards select a UI-language with every payment request, so that international customers have a clue what the device wants them to do. 
This is a huge problem at international events especially with French customers.

**Comment by Angus-Fisher** [comment]
Hi Marcus,

Thanks for the suggestion. At the moment the language is determined based on the region of the merchant. Reader support for the language of the customer's card is on our backlog, but it has not been scheduled so it may still be a little while off. So that we can keep track of which integrators are interested in certain features, could you please tell me you are working with?

Thanks,
Angus

**Comment by MarcusWolschon** [comment]
We write our own software because nobody else did a Magento 2.0 integration.
POS functionality and payments are but a small part.

We do however also often use your Android App standalone and have the same need there. Without any API involved.

**Comment by MarcusWolschon** [comment]
"Reader support for the language of the customer's card is on our backlog, but it has not been scheduled "
Has it been scheduled within the last 7 month?

**Comment by Angus-Fisher** [comment]
Hi Marcus,

Unfortunately there are no changes to date, when we start to tackle this issue we'll inform you here.

Angus

**Comment by MarcusWolschon** [comment]
It's been 2 years and this is still a huge issue for our customers.

**Comment by MarcusWolschon** [comment]
Any update on this issue?

If all else fails, It's not THAT hard to force a different locale on a Context.
(For NumberFormat and DateFormat you have a locale parameter, so that's not a big issue.)

`public void forceLocale(@NonNull final Application application, @NonNull final Locale newLocale) {
        final Resources resources = application.getBaseContext().getResources();
        final Configuration config = resources.getConfiguration();
        config.locale = newLocale;
        resources.updateConfiguration(config, resources.getDisplayMetrics());
    }`

**Comment by jahenry** [comment]
@MarcusWolschon Thanks again for following up here.  Unfortunately this improvement is still in our backlog and we have not yet been able to deliver. We will keep you posted.
Thanks

**Comment by MarcusWolschon** [comment]
So we have 2019 now. Any change?

**Comment by TobiasSuhrborg** [comment]
We would also be interested in this.

**Comment by tavar** [comment]
I would need this feature for my customer, too. He can not understand, why it is not possible. The payment builder would only need an additional parameter.

**Comment by MarcusWolschon** [comment]
> I would need this feature for my customer, too. He can not understand, why it is not possible. The payment builder would only need an additional parameter.

A language parameter?
SumUp knows the language of the customer's card. The app initiating the process doesn't.

**Comment by tavar** [comment]
Yes but the screen is in the device's language. I need to support different languages at once. So the customer can choose this language.

Example:

1. MyDevices is in German.
2. The customer does not under German and presses a button to change the language of myApp to French. 
3. The payment process shown by the sum up SDK shows the screen in German but for the customer expierience, French would be better.

So if there would be a parameter like a language "en", "fr", .... during the SumUpPayment.builder()....               .build(), this would be a great improvement for the end user UX.

**Comment by MarcusWolschon** [comment]
Ok, so as a workaround have the customer manually choose a language instead of reading the language from whatever card the user presents and selecting it automatically because for your app more then just the SumUp payment-process should be in the customer's language.

The app I'm working for is for the merchant and with the SumUp terminals the phone needs to be presented to the customer for a signature after a card payment in some cases.
So the user has already inserted the card and the SumUp app has opened.

**Comment by pati610** [comment]
The suggested feature would be much needed over here in Finland, too.

**Comment by MarcusWolschon** [comment]
Is there at least a way for the API to tell the app what the priority-sorted list of languages (possibly just 1 entry) for the customer is? Probably via a listener-callback or Livedata or intent.

**Comment by michaelprankl99** [comment]
Still not possible? @jahenry

**Comment by ot-repo** [comment]
I really don't believe that this feature is requested 4 years ago and still not implemented. I guess this will never be implemented!

**Comment by MarcusWolschon** [comment]
I think SumUp doesn't know that other languages then English exist on this planet or that dealer regularly sell to people from other countries then their own in-person.

**Comment by pati610** [comment]
I agree, this is unfathomable.

**Comment by r0667002** [comment]
International clients are really confused when they get Dutch instructions on the screen. Please implement this after all.

**Comment by tatianalizarazo** [comment]
At the moment the card readers cannot identify the language of the card used to make a payment. They will only show the language of the country where the SumUp account was registered by default However, it is possible to change the language of the terminals. Please contact our support team so that they can make the change.

**Comment by MarcusWolschon** [comment]
Will the request for a signature on the phone app be shown in the language the bluetooth terminal is set to, if we go through that procedure?

**Comment by fesolet** [comment]
Hi, this is not true, tatianalizarazo, I'm in France, in my computer, my dashboard is in french BUT the app in my phone is in spanish!!  and the tickets is in spanish... my costumers don't understand spanish and it is not possible to change the language. I contact the support team one year ago and it is not possible.

**Comment by jor3l** [comment]
what about adding a language option to SumUpPayment.builder()?

**Comment by YiffyToys** [comment]
> what about adding a language option to SumUpPayment.builder()?

Yes. That's all we're asking for here.

**Comment by leila-sumup** [comment]
Hi everyone, I'm closing this issue. I know that's not what many of you might hope to hear after 9 years, so I want to be straightforward about where we stand on this.

We don't plan to add a language parameter to the SDK or support dynamic language switching on the card reader during a transaction.

The reader's language is set based on the merchant's registered country. It can be changed by contacting our support team, but it's a static setting, not something that can be toggled per transaction.

A few reasons this hasn't moved forward:
-Supporting per-transaction language switching across readers, SDK screens, and receipt flows adds significant complexity for a narrow set of use cases.
-The card reader UI is intentionally minimal (PIN prompt, contactless symbol, amount). For most readers which pair with the SDK, this is largely due to limited screen space. Most of the interaction is icon-based and doesn't require reading text.
-The SDK checkout screens on the phone follow the device's locale, which is the primary point of a user's interaction with text on screen. 

I understand this is frustrating, especially for merchants at international events or in multilingual regions. If this is a blocker for your business, please reach out to integration@sumup.com so we can understand your setup and see if there's a workable path.

Apologies for leaving this open for so long without a definitive answer, we are working now to ensure that all issues get clear and prompt replies.

**Comment by YiffyToys** [comment]
Ah great,
so SumUp stays useless for trade fairs for all merchants who don't have English as their primary language.
Good thing we're already in the process of switching our POS-software to a different provider.

The one told to do a signature on the phone or enter things in the terminal is NOT the user.
It's the customer.
A 3rd party.
And the card reader is the only source giving the customer's primary language or country in a machine readable way.

Android now allowing dynamic, per-application locale changes only works for the POS-application itself
but not the official SumUp app it lauches via the Intent.


A simple button in the SumUp app's checkout screen to change the application-locale
(something Google has added support for long ago) was all that would have been needed.
